test/attr/attr_availability_transitive_osx.swift

```diff
@@ -35,3 +39,70 @@ func osx_extension_call_osx_extension() {
 func osx_extension_call_osx() {
     osx() // expected-error {{'osx()' is unavailable}}
 }
+
+@available(OSX, unavailable)
+var osx_init_osx = osx() // OK
+
+@available(OSXApplicationExtension, unavailable)
+var osx_extension_init_osx = osx() // expected-error {{'osx()' is unavailable}}
+
+@available(OSX, unavailable)
+var osx_inner_init_osx = { let inner_var = osx() } // OK
+
+// FIXME: I'm not sure why this produces two errors instead of just one.
+@available(OSXApplicationExtension, unavailable)
+var osx_extension_inner_init_osx = { let inner_var = osx() } // expected-error 2 {{'osx()' is unavailable}}
+
+struct Outer {
+  @available(OSX, unavailable)
+  var osx_init_osx = osx() // OK
+
+  @available(OSX, unavailable)
+  lazy var osx_lazy_osx = osx() // OK
+
+  @available(OSXApplicationExtension, unavailable)
+  var osx_extension_init_osx = osx() // expected-error {{'osx()' is unavailable}}
+
+  @available(OSXApplicationExtension, unavailable)
+  var osx_extension_lazy_osx = osx() // expected-error {{'osx()' is unavailable}}
+
+  @available(OSX, unavailable)
+  var osx_init_multi1_osx = osx(), osx_init_multi2_osx = osx() // OK
+
+  @available(OSXApplicationExtension, unavailable)
+  var osx_extension_init_multi1_osx = osx(), osx_extension_init_multi2_osx = osx() // expected-error 2 {{'osx()' is unavailable}}
+
+  @available(OSX, unavailable)
+  var (osx_init_deconstruct1_osx, osx_init_deconstruct2_osx) = osx_pair() // OK
+
+  @available(OSXApplicationExtension, unavailable)
+  var (osx_extension_init_deconstruct1_osx, osx_extension_init_deconstruct2_osx) = osx_pair() // expected-error {{'osx_pair()' is unavailable}}
+
+  @available(OSX, unavailable)
+  var osx_inner_init_osx = { let inner_var = osx() } // OK
+  
+  // FIXME: I'm not sure why this produces two errors instead of just one.
+  @available(OSXApplicationExtension, unavailable)
+  var osx_extension_inner_init_osx = { let inner_var = osx() } // expected-error 2 {{'osx()' is unavailable}}
+}
+
+extension Outer {
+  @available(OSX, unavailable)
+  static var also_osx_init_osx = osx() // OK
+  
+  @available(OSXApplicationExtension, unavailable)
+  static var also_osx_extension_init_osx = osx() // expected-error {{'osx()' is unavailable}}
+}
+
+@available(OSX, unavailable)
+extension Outer {
+  static var outer_osx_init_osx = osx() // OK
+}
+
+@available(OSX, unavailable)
+struct NotOnOSX {
+  var osx_init_osx = osx() // OK
+  lazy var osx_lazy_osx = osx() // OK
+  var osx_init_multi1_osx = osx(), osx_init_multi2_osx = osx() // OK
+  var (osx_init_deconstruct1_osx, osx_init_deconstruct2_osx) = osx_pair() // OK
```

Comment:
Just to be sure can you also test a few cases involving patterns like `var (foo, _) = ...`

Reply:
Oh, good call!